perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,773 @@
+  for token in RESPONSE_TIME_TOKENS:
+    logging.info('%s_response_time is %f', token, response_times[token])
+    metric_name = '%s %s' % (SYSBENCH_RESULT_NAME_LATENCY, token)
+
+    if token == 'percentile':
+      metric_name = '%s %s' % (metric_name, FLAGS.sysbench_latency_percentile)
+
+    results.append(sample.Sample(
+        metric_name,
+        response_times[token],
+        MS_UNIT,
+        metadata))
+
+
+def _RunSysbench(vm, metadata):
+  """ Runs the Sysbench OLTP test on the DB instance as indicated by the
+  vm.db_instance_address.
+
+  Args:
+    vm: The client VM that will issue the sysbench test.
+    metadata: The PKB metadata to be passed along to the final results.
+
+  Returns:
+    Results: A list of results of this run.
+  """
+  results = []
+
+  if not hasattr(vm, 'db_instance_address'):
+    logging.error(
+        'Prepare has likely failed, db_instance_address is not found.')
+    raise DBStatusQueryError('RunSysbench: DB instance address not found.')
+
+  # Create the sbtest database for Sysbench.
+  create_sbtest_db_cmd = ('mysql -h %s -u %s -p%s '
+                          '-e \'create database sbtest;\'') % (
+                              vm.db_instance_address,
+                              MYSQL_ROOT_USER,
+                              MYSQL_ROOT_PASSWORD)
+  stdout, stderr = vm.RemoteCommand(create_sbtest_db_cmd)
+  logging.info('sbtest db created, stdout is %s, stderr is %s',
+               stdout, stderr)
+
+  # Provision the Sysbench test based on the input flags (load data into DB)
+  # Could take a long time if the data to be loaded is large.
+  data_load_start_time = time.time()
+  data_load_cmd_tokens = ['sysbench',
+                          '--test=%s' % PREPARE_SCRIPT_PATH,
+                          '--oltp_tables_count=%d' % FLAGS.oltp_tables_count,
+                          '--oltp-table-size=%d' % FLAGS.oltp_table_size,
+                          '--rand-init=%s' % RAND_INIT_ON,
+                          '--num-threads=%d' % FLAGS.oltp_tables_count,
+                          '--mysql-user=%s' % MYSQL_ROOT_USER,
+                          '--mysql-password="%s"' % MYSQL_ROOT_PASSWORD,
+                          '--mysql-host=%s' % vm.db_instance_address,
+                          'run']
+  data_load_cmd = ' '.join(data_load_cmd_tokens)
+
+  # Sysbench output is in stdout, but we also get stderr just in case
+  # something went wrong.
+  stdout, stderr = vm.RobustRemoteCommand(data_load_cmd)
+  load_duration = time.time() - data_load_start_time
+  logging.info('It took %d seconds to finish the data loading step',
+               load_duration)
+  logging.info('data loading results: \n stdout is:\n%s\nstderr is\n%s',
+               stdout, stderr)
+
+  results.append(sample.Sample(
+      SYSBENCH_RESULT_NAME_DATA_LOAD,
+      load_duration,
+      SECONDS_UNIT,
+      metadata))
+
+  # Now run the sysbench OLTP test and parse the results.
+
+  for i in xrange(1, 3):
```

Comment:
Done. Refactored some logic from here into a function "_IssueSysbenchCommand()" and then we call the function from this loop.